test/e2e/infinispan/storage_test.go

```diff
@@ -28,8 +27,9 @@ func TestNodeWithStorageClass(t *testing.T) {
 
 	// Ensure a PVCs is bound from defaultStorageClass
 	pvcName := "data-volume-test-node-with-storage-class-0"
-	if *testKube.GetPVC(pvcName, spec.Namespace).Spec.StorageClassName != defaultStorageClass {
-		tutils.ExpectNoError(fmt.Errorf("persistent volume claim (%s) was created, but not bound to the cluster's default storage class (%s)",
-			pvcName, defaultStorageClass))
+	storageClassName := testKube.GetPVC(pvcName, spec.Namespace).Spec.StorageClassName
+	if (defaultStorageClass == "" && storageClassName != nil) || (defaultStorageClass != "" && *storageClassName != defaultStorageClass) {
```

Comment:
Just an idea, might be easier to read:
```
if (defaultStorageClass == "") {
  require.Nil(storageClassName, "StorageClassName should be empty")
} else {
  require.Equal(defaultStorageClass, storageClassName, "StorageClassName should use default storage class")
}
```
Maybe we could go even step further by using  only require.Equal() but I'm not sure what happens on `""` vs `nil` comparison